imap/mboxlist.c

```diff
@@ -1770,129 +1813,22 @@ static int mboxlist_createmailbox_full(const char *mboxname, int mbtype,
     return r;
 }
 
-EXPORTED int mboxlist_createmailboxlock(const char *name, int mbtype,
-                           const char *partition,
-                           int isadmin, const char *userid,
-                           const struct auth_state *auth_state,
-                           int localonly, int forceuser, int dbonly,
-                           int notify, struct mailbox **mailboxptr)
+EXPORTED int mboxlist_createmailboxlock(const mbentry_t *mbentry,
+                                        unsigned options, modseq_t highestmodseq,
+                                        unsigned isadmin, const char *userid,
+                                        const struct auth_state *auth_state,
+                                        unsigned flags, struct mailbox **mboxptr)
 {
-    struct mboxlock *namespacelock = mboxname_usernamespacelock(name);
+    struct mboxlock *namespacelock = mboxname_usernamespacelock(mbentry->name);
```

Comment:
I expect we'll want to do mbentry_usernamespacelock(mbentry) at some point :)  And lock by uniqueid.